contract/test/Custody.t.sol

```diff
@@ -676,49 +676,42 @@ contract CustodyTest is Test {
         vm.prank(guest);
         custody.join(channelId, 1, guestSig);
 
-        // 2. Create a final state for closing
-        State memory finalState = createClosingState();
-
-        // Both sign the final state
-        Signature memory hostFinalSig = signState(chan, finalState, hostPrivKey);
-        Signature memory guestFinalSig = signState(chan, finalState, guestPrivKey);
-
-        Signature[] memory finalSigs = new Signature[](2);
-        finalSigs[0] = hostFinalSig;
-        finalSigs[1] = guestFinalSig;
-        finalState.sigs = finalSigs;
+        // 2. Create a resize state with CHANRESIZE magic number
+        State memory resizeState = initialState;
 
-        // 3. Create new channel config with different nonce
-        address[] memory participants = new address[](2);
-        participants[0] = host;
-        participants[1] = guest;
+        // Create resize data with magic number and resize amounts
+        int256[] memory resizeAmounts = new int256[](2);
+        resizeAmounts[0] = int256(DEPOSIT_AMOUNT); // Increase host's deposit by DEPOSIT_AMOUNT
+        resizeAmounts[1] = -int256(DEPOSIT_AMOUNT/2); // Decrease guest's deposit by DEPOSIT_AMOUNT/2
+        resizeState.data = abi.encode(CHANRESIZE, resizeAmounts);
 
-        Channel memory newChan = Channel({
-            participants: participants,
-            adjudicator: address(adjudicator),
-            challenge: CHALLENGE_DURATION,
-            nonce: NONCE + 1
-        });
+        // Update allocations to match the resize
+        resizeState.allocations[0].amount = DEPOSIT_AMOUNT * 2; // Host now has 2x initial amount
+        resizeState.allocations[1].amount = DEPOSIT_AMOUNT / 2; // Guest now has half initial amount
 
-        // 4. Create new initial state for the new channel
-        State memory newInitialState = createInitialState();
+        // Both sign the resize state
+        Signature memory hostResizeSig = signState(chan, resizeState, hostPrivKey);
+        Signature memory guestResizeSig = signState(chan, resizeState, guestPrivKey);
 
-        // Host signs new initial state
-        Signature memory hostNewSig = signState(newChan, newInitialState, hostPrivKey);
-        Signature[] memory newHostSigs = new Signature[](1);
-        newHostSigs[0] = hostNewSig;
-        newInitialState.sigs = newHostSigs;
+        Signature[] memory resizeSigs = new Signature[](2);
+        resizeSigs[0] = hostResizeSig;
+        resizeSigs[1] = guestResizeSig;
+        resizeState.sigs = resizeSigs;
 
-        // 5. Reset the channel
+        // 3. Resize the channel
         vm.prank(host);
-        custody.reset(channelId, finalState, new State[](0), newChan, newInitialState);
+        custody.resize(channelId, resizeState);
```

Comment:
after resize, and without closing can we test if Charlie can withdraw the full unlocked amount?